### PR TITLE
Package files

### PR DESCRIPTION
**Pull request by livarcocc**
Fixes https://github.com/dotnet/cli/issues/4338

@brthor @piotrpMSFT @eerhardt @jgoshi

### CODE REVIEW

**Comment by brthor** [comment]
cc @rohit21agrawal

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackOptionsRule.cs [hunk 1]
```diff
@@ -52,6 +56,19 @@ public class MigratePackOptionsRule : IMigrationRule
                     packOptions => packOptions.RepositoryUrl,
                     packOptions => !string.IsNullOrEmpty(packOptions.RepositoryUrl));
 
+        private IncludeContextTransform PackFilesTransform =>
+            new IncludeContextTransform("Content", transformMappings: true)
+                .WithMetadata("Pack", "True")
```
Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackOptionsRule.cs]: I think true is lowercase here

File: src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackOptionsRule.cs [hunk 2]
```diff
@@ -80,21 +97,52 @@ private void ConstructTransformLists()
...
+            }
+
+            return Path.GetDirectoryName(targetPath);
+        }
+
+        private bool PathIsDirectory(string targetPath)
+        {
+            var normalizedTargetPath = PathUtility.GetPathWithDirectorySeparator(targetPath);
+
+            return normalizedTargetPath[normalizedTargetPath.Length - 1] == Path.DirectorySeparatorChar;
```
Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigratePackOptionsRule.cs]: does this work with forward slashes on windows?

File: src/Microsoft.DotNet.ProjectJsonMigration/transforms/IncludeContextTransform.cs [hunk 3]
```diff
@@ -84,6 +93,13 @@ public IncludeContextTransform WithMetadata(string metadataName, Func<IncludeCon
             return this;
         }
 
+        public IncludeContextTransform WithMappingsToTransform(
+            Func<AddItemTransform<IncludeContext>, string, AddItemTransform<IncludeContext>> mappingsToTransfrom)
+        {
+            _mappingsToTransfrom = mappingsToTransfrom;
```
Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/transforms/IncludeContextTransform.cs]: `_mappingsMetadataTransform` ?

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs [hunk 4]
```diff
@@ -187,19 +187,75 @@ public void Migrating_Repository_Url_populates_RepositoryUrl()
...
         [Fact]
-        public void Migrating_Files_throws_an_exception_for_now()
+        public void Migrating_Files_without_mappings_populates_content_with_same_path_as_include_and_pack_true()
         {
-            Action action = () => RunPackOptionsRuleOnPj(@"
+            var mockProj = RunPackOptionsRuleOnPj(@"
+                {
+                    ""packOptions"": {
+                        ""files"": {
+                            ""include"": [""path/to/some/file.cs"", ""path/to/some/other/file.cs""]
```
Comment (brthor) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs]: test this with an absolute path as well?

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs [hunk 5]
```diff
@@ -187,19 +187,75 @@ public void Migrating_Repository_Url_populates_RepositoryUrl()
...
+                        }
+                    }
+                }");
+
+            var contentItems = mockProj.Items
+                .Where(item => item.ItemType.Equals("Content", StringComparison.Ordinal))
+                .Where(item => item.GetMetadataWithName("Pack").Value == "True");
```
Comment (brthor) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs]: lowercase true

Comment (livarcocc) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs]: Fixed.

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs (hunk 4 above)
Comment (livarcocc) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs]: Added.

File: src/Microsoft.DotNet.ProjectJsonMigration/transforms/IncludeContextTransform.cs (hunk 3 above)
Comment (brthor) [inline src/Microsoft.DotNet.ProjectJsonMigration/transforms/IncludeContextTransform.cs]: agreed, I can't think of something better

File: test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs (hunk 4 above)
Comment (livarcocc) [inline test/Microsoft.DotNet.ProjectJsonMigration.Tests/Rules/GivenThatIWantToMigratePackOptions.cs]: Actually, this is not allowed in pj world.

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x64 Release Build

**Comment by brthor** [comment]
test Windows_NT x64 Release Build
test RHEL7.2 x64 Release Build
test OSX x64 Release Build

**Pull request merged**